yt_dlp/extractor/bilibili.py

```diff
@@ -774,3 +775,119 @@ def _real_extract(self, url):
+        for episode in data_json.get('episodes', []):
+            if str(episode.get('ep_id')) == id:
+                episode_data = episode
+                break
+        video_json = self._download_json(f'{api_url}/web/playurl?ep_id={id}&platform=web', id)['data']
+        if not video_json:
+            self.raise_login_required(method='cookies')
+        video_json = video_json['playurl']
+        formats = []
+        for vid in video_json.get('video', []):
+            video_res = vid['video_resource']
+            video_info = vid['stream_info']
+            if video_res.get('url'):
+                formats.append({
+                    'url': video_res['url'],
+                    'ext': 'mp4',
+                    'format_note': video_info.get('desc_words'),
+                    'width': video_res.get('width'),
+                    'height': video_res.get('height'),
+                    'vbr': video_res.get('bandwidth'),
+                    'acodec': 'none',
+                    'vcodec': video_res.get('codecs'),
+                    'filesize': video_res.get('size'),
+                })
+        for aud in video_json.get('audio_resource', []):
+            if aud.get('url'):
+                formats.append({
+                    'url': aud['url'],
+                    'ext': 'mp4',
+                    'abr': aud.get('bandwidth'),
+                    'acodec': aud.get('codecs'),
+                    'vcodec': 'none',
+                    'filesize': aud.get('size'),
+                })
+        sub_json = self._download_json(f'{api_url}/m/subtitle?ep_id={id}&platform=web', id)['data']
+        subtitles = {}
+        for sub in sub_json.get('subtitles', []):
+            sub_url = sub.get('url')
+            if not sub_url:
+                continue
+            subtitles.setdefault(sub.get('key', 'en'), []).append({
+                'url': sub_url,
+            })
+
+        self._sort_formats(formats)
+
+        return {
+            'id': id,
+            'title': episode_data['long_title'] or episode_data.get('title'),
+            'thumbnail': episode_data.get('cover'),
+            'episode_number': str_to_int(episode_data.get('title')),
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+
+class BiliIntlSeriesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bili(?P<type>bili\.tv|intl.com)/(?:[a-z]{2}/)?play/(?P<id>\d+)$'
+    _TESTS = [{
+        'url': 'https://www.bilibili.tv/en/play/1005144',
+        'playlist_mincount': 220,
+        'info_dict': {
+            'id': '1005144',
+        },
+    }, {
+        'url': 'https://www.biliintl.com/en/play/34613',
+        'playlist_mincount': 15,
+        'info_dict': {
+            'id': '34613',
+        },
+    }]
+
+    def _entries(self, id, type):
+        data_json = self._download_json(f'https://api.bili{type}/intl/gateway/web/view/ogv_collection?season_id={id}', id)['data']
```

Comment:
When downloading a series, this API is redownloaded for each episode. Would be better if the seriesIE itself can extract everything needed